content/docs/testing-recipes.md

```diff
@@ -95,46 +78,42 @@ export default function User(props) {
 We can write tests for it:
 
 ```jsx
+// user.test.js
+import { render } from "@testing-library/react";
 import User from "./user";
-import { render } from "react-dom";
-import { act } from "react-dom/test-utils";
 
-let container = null
 beforeEach(() => {
-  // setup a DOM element as a render target
-  container = document.createElement('div')
-  document.body.appendChild(container)
-})
+  // mock window.fetch so the unit tests don't actually make HTTP requests
+  jest.spyOn(window, "fetch").mockImplementation(() => {});
+});
 
 afterEach(() => {
-  // cleanup on exiting
-  container.remove();
-  container = null  
-})
+  // remove the mock to ensure tests are completely isolated
+  window.fetch.mockRestore();
```

Comment:
This needs to be in an `afterEach` to avoid cleanup bugs (learn more in the "What about cleanup?" section of https://kentcdodds.com/blog/avoid-nesting-when-youre-testing)

Reply:
+1 cleanup actually requires a fair amount of care to get right. If you mess it up you get cross-test coupling and false positives.

https://github.com/reactjs/reactjs.org/pull/2182

Reply:
Exactly. Thanks for bringing that up Alex. So this PR is about more than just "remove the boilerplate" but it's also "avoid people making mistakes when writing the boilerplate."